chapter3/challenges/program/lib/password_manager_2.rb

```diff
@@ -1,2 +1,57 @@
 class PasswordManager2
-end
\ No newline at end of file
+  def initialize
+    #Initializes three arrays, password manager being the main one, and two other arrays for checking / ease of ordering.
+    @password_manager = Array.new()
+    @password_array = Array.new()
+    @service_array = Array.new()
+  end
+
+  def add(service, password)
+    if @service_array.include? service
+      return "ERROR: Service names must be unique"
+    elsif @password_array.include? password
+      return "ERROR: Passwords must be unique"
+    else
+        new_entry = {
+            "service" => service,
+            "password" => password,
+            "added" => DateTime.now(),
+          }
+      @password_manager << new_entry
+      @service_array << service
+      @password_array << password
+    end
+  end
+
+  def services
+    return @service_array
+  end
+
+  def remove(service)
+    deleted_pass = @password_manager.find { |h| h["service"] == service }["password"]
+    @password_manager.map! { |e| e[service] != service }
```

Comment:
I don't see why this is .map!
In irb, with arr = [{1=>2, 2=>3, 3=>4}, {51=>2, 2=>3, 3=>4}] 
arr.map {|el| el[1] != 2 }
 => [false, true]

Reply:
Yeah you're right, I've fixed it now by using select! instead. Thank you.